src/Compilers/CSharp/Test/Emit3/ExtensionTypeTests.cs

```diff
@@ -127,6 +130,28 @@ private static void AssertSetStrictlyEqual(string[] expected, string[] actual)
             Assert.False(sourceNamedType.IsImplicitlyDeclared);
         }
 
+        if (type is PENamedTypeSymbol peType)
+        {
+            var module = (PEModuleSymbol)type.ContainingModule;
+            var reader = module.Module.GetMetadataReader();
+            var fieldDefHandle = reader.GetTypeDefinition(peType.Handle).GetFields()
+                .Where(f => reader.GetString(reader.GetFieldDefinition(f).Name) == WellKnownMemberNames.ExtensionFieldName).SingleOrDefault();
+
+            // Static extensions don't have this field, but non-static extensions have it
+            Assert.Equal(fieldDefHandle.IsNil, type.IsStatic);
+
+            if (!type.IsStatic)
+            {
+                var fieldDef = reader.GetFieldDefinition(fieldDefHandle);
```

Comment:
>var fieldDef = reader.GetFieldDefinition(fieldDefHandle); [](http://example.com/codeflow?start=16&length=57)

It feels like it would be easier to crate an instance of PEFieldSymbol here and validate it instead.

Reply:
Thanks. That is a nice simplification indeed